src/main/scala/com/fulcrumgenomics/umi/CorrectUmis.scala

```diff
@@ -198,6 +203,11 @@ class CorrectUmis
 
             // Output the corrected read
             if (matches.forall(_.matched)) {
+              // Store the original UMI if requested, and there are mismatches
```

Comment:
(1) and **if** there are mismatches
(2) if both requested and there are mismatches